### PR TITLE
[T4A5][F09-B1] Phua Kian Ming

### PR DESCRIPTION
**Pull request by daredevil999**
Added an association class Tagging to store the list of added and deleted tags in a session. Upon exiting, the Tagging list will be printed.

### CODE REVIEW

**Comment by daredevil999** [comment]
Ready for review

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        StringBuilder builder = new StringBuilder();
+        if (action == Action.ADD) {
+            builder.append(ADD_ACTION);
+        } else {
+            builder.append(DELETE_ACTION);
+        }
+        builder.append(personName)
+                .append(" ")
+                .append(tag.toString());
+        return builder.toString();
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/tag/Tagging.java:38]: You can also take a look at `String.join()` method.

File: src/seedu/addressbook/data/AddressBook.java [hunk 2]
```diff
@@ -129,6 +180,19 @@ public void clear() {
...
+     */
+    public String exitTaggingList() {
+        StringBuilder builder =  new StringBuilder();
+        for (Tagging t : tagging) {
+            builder.append(t.toString())
+                    .append("\n");
+        }
+        tagging.clear();
+        return builder.toString();
+    }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:195]: Good job for adding above methods.

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -39,14 +45,43 @@ public AddressBook() {
...
+    public static ArrayList<Tagging> addExistingTags(UniquePersonList persons) {
+        ArrayList<Tagging> taggings = new ArrayList<Tagging>();
+        for (Person p: persons ) {
+            UniqueTagList individualTagList = p.getTags();
+            for (Tag t: individualTagList) {
+                taggings.add(new Tagging(Action.ADD, p.getName().toString(), t));
+            }
+        }
+        return taggings;
+    }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:82]: Maybe you can come up with better abstraction here, remember SLAP.

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package seedu.addressbook.data.tag;
+
+/**
+ * An Association class to keep track of added or deleted tags of all persons in the address book.
+ * 
+ */
+
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/tag/Tagging.java:7]: Good job for adding the header comment.

**Comment by jeffryhartanto** [comment]
Some comments added. Please close the PR after reading comments. Overall I'm satisfied with your work, you add the header comment, maintaining encapsulation and also adding several methods for tagging in the AddressBook class. Keep up the good work.

**Comment by daredevil999** [comment]
Thanks! sorry for the late submission...

**Pull request closed without merging**